fix: Sanitize filename on download and correction of filename header

oscardimanno opened the pull request:
This PR fixes issue #9905 .

Currently, Roundcube chooses whether to use the filename or filename* parameter based on the characters in the filename.

This PR simplifies the logic behind this.
According to [RFC 6266, section 4.3](https://datatracker.ietf.org/doc/html/rfc6266#section-4.3), and the [MDN documentation](https://developer.mozilla.org/en-US/docs/Web/HTTP/Reference/Headers/Content-Disposition#syntax), the user agent prioritises the 'filename*' parameter if it is present and ignores the 'filename' parameter.
This change aligns the code with the IETF standard while maintaining compatibility with older clients.

alecpl commented:
I think using both parameters is fine and the way to go. Please, fix tests.

alecpl commented on program/lib/Roundcube/rcube_output.php:
```diff
@@ -256,17 +256,15 @@ public function download_headers($filename, $params = [])
+            if (!empty($params['charset']) && rcube_charset::is_valid($params['charset'])) {
+                $charset = $params['charset'];
             }
+            $disposition .= "; filename*={$charset}''{$filename}";
```
Maybe it would make sense to add this parameter only when `$fallback_filename != $filename`.

oscardimanno commented:
Hi @alecpl,
closing this PR.
I've cleaned up my master branch and moved the changes on a dedicated branch to resolve a CI issue.
The new, updated pull request is here: #9960
